### DISCUSSION TITLE
How to enable access to the microphone?

### DISCUSSION THREAD

**Discussion by drupol** [open-webui/open-webui#3012] [Q&A] [answered]
Hello,

When I try to use the microphone, I get a permission denied.

Open-WebUI is installed on a local server, not using HTTPS (yet), could it be the root cause of the issue? I also checked in Firefox if there were permission related to microphone, but couldn't find anything relevant. I also tested with Chromium, same issue.

![image](https://github.com/open-webui/open-webui/assets/252042/999f5056-104c-43ae-8ce0-b255f0c06451)

![image](https://github.com/open-webui/open-webui/assets/252042/dbd2e6a3-e08b-40d5-87ee-e184eaee4a38)

Thanks!

**Comment by JuanjoJG-WATA** [comment, marked as answer]
You need https for that. If you use localhost (the server being in the same machine as the one you access the WebUI), you can use it, because most browsers see localhost as a safe address, but if you access remotely, you'll need to be using https.

**Reply by drupol** [reply]
OK good, I'll setup Caddy in front of it right now. Thanks!

**Reply by drupol** [reply]
That works pretty well indeed!

**Reply by JuanjoJG-WATA** [reply]
I'm glad that made it work!

**Reply by Edward205** [reply]
> That works pretty well indeed!

Hello, can you share your Caddy configuration? I've set up Caddy with HTTPS as well but I'm not able to use the microphone due to the same error. Perhaps I made a mistake, because I never used Caddy before, and I just followed the simple documentation. Thanks!
![image](https://github.com/open-webui/open-webui/assets/52004020/c101e0dd-cade-48df-9f69-29abb2a382e4)

**Reply by justinh-rahb** [reply]
Caddy prides itself on their very secure default configuration. It is possible there is a content security policy being applied that is blocking media devices.

**Reply by JuanjoJG-WATA** [reply]
@Edward205 try it the old fashion way. Create your own self-signed certificates, add them to start.sh and change the port to 443. That did the trick (temporarily) for me. I can give you some indications if you need help.

**Reply by Edward205** [reply]
@justinh-rahb @JuanjoJG-WATA 
I did some debugging and I found that the error was caused due to the fact that I didn't have a video device plugged in. This line was causing the error:
https://github.com/open-webui/open-webui/blob/95ff355737b78f1e50e8151cb0c1728cda96eae3/src/lib/components/chat/MessageInput.svelte#L938
From my understanding, this code demands access to both the microphone and camera, and if either one fails to give access, the error is shown. I'm not a Javascript expert, but if someone wants to look into it, this may be unintended behaviour. If I misunderstood the code, and this was intentional, then I apologise. 
So I plugged in a webcam and I acessed OpenWebUI via localhost (without HTTPS). It works! 
Thank you for the help!

**Reply by JuanjoJG-WATA** [reply]
Yeah, you are the localhost, so browsers consider it safe and will trust any device. The problem comes when you try to access the WebUI remotely, lets say your installation is in a remote server and your need to connect to it through the IP **192.168.1.100:8080**, for example.

In this case, unless you have an active https connection, the browser wont let you use the microphone or camera, as it considers them not safe.

**Reply by thiswillbeyourgithub** [reply]
> You need https for that. If you use localhost (the server being in the same machine as the one you access the WebUI), you can use it, because most browsers see localhost as a safe address, but if you access remotely, you'll need to be using https.


Actually this is not true anymore. I made [a PR to the docs](https://github.com/open-webui/docs/pull/172) to explain how to whitelist a URL in chrome based browsers: 

>Basically:

    Open chrome://flags/#unsafely-treat-insecure-origin-as-secure
    Add your local ip there: 192.168.1.39:3000 for example
    Restart the browser

**Reply by ctxadm** [reply]
thats it, thanks !!!

**Comment by yaobohan888** [comment]
出错了 ：/ 无法在模型 'D：\AI\Products\Ollama_OpenWebUI_2_8_3_10\open-webui\venv\Lib\site-packages\data/cache/whisper/models\models--Systran--faster-whisper-base\snapshots\ebe41f70d5b6dfa9166e2c581c45c9c0cfc57b66' 中打开文件“model.bin”

**Comment by yaobohan888** [comment]
edge://flags/#edge-automatic-https    找到Insecure origins treated as secure
Treat given (insecure) origins as secure origins. Multiple origins can be supplied as a comma-separated list. Origins must have their protocol specified e.g. "http://example.com". For the definition of secure contexts, see https://w3c.github.io/webappsec-secure-contexts/ – Mac, Windows, Linux, Android   改成启用。填写http://自己地址，可以启用麦克风，但是还是出现了远程错误

**Comment by SidSaxena** [comment]
What is the firefox equivalent of changing the insecure origin flag?

**Reply by natea** [reply]
@SidSaxena this has been documented here: https://docs.openwebui.com/troubleshooting/microphone-error/#firefox-based-browsers
